### PR TITLE
Migrate docs to use Intellij instead of Eclipse #200

### LINKED ISSUE #200: Migrate docs to use Intellij instead of Eclipse
Similar to level 1

### PR DESCRIPTION
**Pull request by limmlingg**
Fixes #200

### CODE REVIEW

**Comment by CanIHasReview** [comment]
# v1

@limmlingg submitted v1 for review.

* [v1 1/1] [Convert docs from Eclipse to IntelliJ](https://github.com/se-edu/addressbook-level2/pull/201/commits/53b665ec7b016aee091fa644b890d4d72bfcd98a)

([:books: Archive](https://github.com/se-edu/addressbook-level2/compare/1603ad0271fc873b66ca35d4f1a922d3e199574a...53b665ec7b016aee091fa644b890d4d72bfcd98a))
<details>
<summary>Checkout this PR version locally</summary>

```sh
git fetch https://github.com/se-edu/addressbook-level2.git refs/pr/201/1/head:BRANCHNAME
```
where `BRANCHNAME` is the name of the local branch you wish to fetch this PR to.

</details>

File: doc/DeveloperGuide.md [hunk 1]
```diff
@@ -5,16 +5,21 @@
...
-5. Click `Finish`
+1. Open IntelliJ (if you are not in the welcome screen, click `File` > `Close Project` to close the existing project dialog first)
+2. Set up the correct JDK version
+   1. Click `Configure` > `Project Defaults` > `Project Structure`
+   2. Click `New...` and find the directory of the JDK
+3. Click `Import Project`
+4. Locate the project directory and click `OK`
+5. Select `Create project from existing sources` and click `Next`
+6. Rename the project if you want. Click `Next`
+7. Ensure that your src folder is checked. Keep clicking `Next`
```
Comment (yamgent) [inline doc/DeveloperGuide.md]: There are two folders in the list on my screen. Do I need to check both `\src` and `\test\java`?

File: doc/DeveloperGuide.md [hunk 2]
```diff
@@ -39,7 +44,7 @@
 
 ### JUnit tests
 
-* In Eclipse, right-click on the `test/java` folder and choose `Run as` > `JUnit Test`
+* In IntelliJ, right-click on the `test/java` folder and choose `Run 'All Tests'`
```
Comment (yamgent) [inline doc/DeveloperGuide.md]: Didn't work, no such thing on my screen :(.

screenshot

File: doc/UserGuide.md [hunk 3]
```diff
@@ -5,20 +5,21 @@ Please refer to the [Setting up](DeveloperGuide.md#setting-up) section to learn
...
 
-**Using Eclipse**
+**Using IntelliJ**
 
-1. Find the project in the `Project Explorer` or `Package Explorer` (usually located at the left side)
-2. Right click on the project
-3. Click `Run As` > `Java Application`
+1. Find the project in the `Project Explorer` (usually located at the left side)
+   1. If the `Project Explorer` is not visible, press <kbd>ALT</kbd>+<kbd>1</kbd> for Windows/Linux, <kbd>CMD</kbd>+<kbd>1</kbd> for macOS to open the `Project Explorer` tab
+2. Go to the `src` folder and locate the `AddressBook` file
```
Comment (yamgent) [inline doc/UserGuide.md]: ... `Main` file inside `seedu.addressbook`?

File: doc/UserGuide.md [hunk 4]
```diff
@@ -5,20 +5,21 @@ Please refer to the [Setting up](DeveloperGuide.md#setting-up) section to learn
...
+3. Right click the file and select `Run AddressBook.main()`
```
Comment (yamgent) [inline doc/UserGuide.md]: `Run Main.main()`

File: doc/UserGuide.md [hunk 5]
```diff
@@ -5,20 +5,21 @@ Please refer to the [Setting up](DeveloperGuide.md#setting-up) section to learn
...
 5. Now you can interact with the program through the `Console`
 
 **Using Command Line**
 
-1. 'Build' the project using Eclipse
+1. 'Build' the project using IntelliJ
 2. Open the `Terminal`/`Command Prompt`
-3. `cd` into the project's `bin` directory
-4. Type `java seedu.addressbook.Main`, then <kbd>Enter</kbd> to execute
+3. `cd` into the project's `out\production\addressbook-level1` directory
```
Comment (yamgent) [inline doc/UserGuide.md]: `addressbook-level2`.

File: doc/UserGuide.md [hunk 6]
```diff
@@ -5,20 +5,21 @@ Please refer to the [Setting up](DeveloperGuide.md#setting-up) section to learn
...
+4. Type `java seedu.addressbook.AddressBook`, then <kbd>Enter</kbd> to execute
```
Comment (yamgent) [inline doc/UserGuide.md]: `java seedu.addressbook.Main`.

**Comment by CanIHasReview** [comment]
# v2

@limmlingg submitted v2 for review.

* [v2 1/1] [Convert docs from Eclipse to IntelliJ](https://github.com/se-edu/addressbook-level2/pull/201/commits/76d69215691d527e9d2d37b708afb97ae84a8866)

([:books: Archive](https://github.com/se-edu/addressbook-level2/compare/1603ad0271fc873b66ca35d4f1a922d3e199574a...76d69215691d527e9d2d37b708afb97ae84a8866)) ([:chart_with_upwards_trend: Interdiff between v1 and v2](https://github.com/se-edu/addressbook-level2/compare/4e7c3a447abb6efbe1c257e21c814dafe030367e~1...4e7c3a447abb6efbe1c257e21c814dafe030367e))
<details>
<summary>Checkout this PR version locally</summary>

```sh
git fetch https://github.com/se-edu/addressbook-level2.git refs/pr/201/2/head:BRANCHNAME
```
where `BRANCHNAME` is the name of the local branch you wish to fetch this PR to.

</details>

**Comment by CanIHasReview** [comment]
# v3

@limmlingg submitted v3 for review.

* [v3 1/1] [Convert docs from Eclipse to IntelliJ](https://github.com/se-edu/addressbook-level2/pull/201/commits/fb1a714933a856a7430cac52c42a014e699f6226)

([:books: Archive](https://github.com/se-edu/addressbook-level2/compare/1603ad0271fc873b66ca35d4f1a922d3e199574a...fb1a714933a856a7430cac52c42a014e699f6226)) ([:chart_with_upwards_trend: Interdiff between v2 and v3](https://github.com/se-edu/addressbook-level2/compare/df609eb9616f9176faea985eaac7e0ae11397523~1...df609eb9616f9176faea985eaac7e0ae11397523))
<details>
<summary>Checkout this PR version locally</summary>

```sh
git fetch https://github.com/se-edu/addressbook-level2.git refs/pr/201/3/head:BRANCHNAME
```
where `BRANCHNAME` is the name of the local branch you wish to fetch this PR to.

</details>

File: doc/DeveloperGuide.md [hunk 7]
```diff
@@ -5,16 +5,25 @@
...
+   1. Click `Configure` > `Project Defaults` > `Project Structure`
+   2. Click `New...` and find the directory of the JDK
+3. Click `Import Project`
+4. Locate the project directory and click `OK`
+5. Select `Create project from existing sources` and click `Next`
+6. Rename the project if you want. Click `Next`
+7. Ensure that your `\src` and `\test\java` folders are checked. Keep clicking `Next`
+8. Click `Finish`
+9. Add JUnit 4 to classpath
+   1. Open any test file in `\test\java` and place your cursor over any `@test` highlighted in red
```
Comment (yamgent) [inline doc/DeveloperGuide.md]: Should be `@Test`.

File: doc/DeveloperGuide.md [hunk 8]
```diff
@@ -39,7 +48,8 @@
 
 ### JUnit tests
 
-* In Eclipse, right-click on the `test/java` folder and choose `Run as` > `JUnit Test`
+* In IntelliJ, right-click on the `test` folder and choose `Run 'All Tests'`
+* If `StorageFileTest` throws `NoSuchFileException`, click the `All in test` drop down menu at the top right corner of the window. Select `Edit configurations...`. Under `Configuration`, change the `Working directory` to the `addressbook-level2` folder
```
Comment (yamgent) [inline doc/DeveloperGuide.md]: Should be a separate info box on its own (reason: it is only done once and it is "additional" information).

**Comment by CanIHasReview** [comment]
# v4

@limmlingg submitted v4 for review.

* [v4 1/1] [Convert docs from Eclipse to IntelliJ](https://github.com/se-edu/addressbook-level2/pull/201/commits/8a346784e764e5ab0dfaf0da73d18d25c22fbd02)

([:books: Archive](https://github.com/se-edu/addressbook-level2/compare/1603ad0271fc873b66ca35d4f1a922d3e199574a...8a346784e764e5ab0dfaf0da73d18d25c22fbd02)) ([:chart_with_upwards_trend: Interdiff between v3 and v4](https://github.com/se-edu/addressbook-level2/compare/ac829746dcc745d7a9ce886f38b68dedad18f447~1...ac829746dcc745d7a9ce886f38b68dedad18f447))
<details>
<summary>Checkout this PR version locally</summary>

```sh
git fetch https://github.com/se-edu/addressbook-level2.git refs/pr/201/4/head:BRANCHNAME
```
where `BRANCHNAME` is the name of the local branch you wish to fetch this PR to.

</details>

File: doc/DeveloperGuide.md [hunk 9]
```diff
@@ -39,7 +48,8 @@
 
 ### JUnit tests
 
-* In Eclipse, right-click on the `test/java` folder and choose `Run as` > `JUnit Test`
+* In IntelliJ, right-click on the `test` folder and choose `Run 'All Tests'`
+> If `StorageFileTest` throws `NoSuchFileException`, click the `All in test` drop down menu at the top right corner of the window. Select `Edit configurations...`. Under `Configuration`, change the `Working directory` to the `addressbook-level2` folder
```
Comment (yamgent) [inline doc/DeveloperGuide.md]: One last nit, maybe put numbered list? :P

**Comment by CanIHasReview** [comment]
# v5

@limmlingg submitted v5 for review.

* [v5 1/1] [Convert docs from Eclipse to IntelliJ](https://github.com/se-edu/addressbook-level2/pull/201/commits/527b3f12066fa6fceebc54248298bb15dbf9adf5)

([:books: Archive](https://github.com/se-edu/addressbook-level2/compare/1603ad0271fc873b66ca35d4f1a922d3e199574a...527b3f12066fa6fceebc54248298bb15dbf9adf5)) ([:chart_with_upwards_trend: Interdiff between v4 and v5](https://github.com/se-edu/addressbook-level2/compare/2f47087c23751b689e0009044c656be587abc4c6~1...2f47087c23751b689e0009044c656be587abc4c6))
<details>
<summary>Checkout this PR version locally</summary>

```sh
git fetch https://github.com/se-edu/addressbook-level2.git refs/pr/201/5/head:BRANCHNAME
```
where `BRANCHNAME` is the name of the local branch you wish to fetch this PR to.

</details>

**Comment by limmlingg** [comment]
@yamgent Actually, should the StorageFileTest failure be documented under `Troubleshooting test failures` instead?

**Comment by yamgent** [comment]
> @yamgent Actually, should the StorageFileTest failure be documented under Troubleshooting test failures instead?

Hmm...

* It is technically a failure of a test case, so it seems appropriate to put in under a section that mentions about test case failures.
* However, it is not really a "troubleshooting" because this will most likely happen to everyone.

File: doc/DeveloperGuide.md [hunk 10]
```diff
@@ -39,7 +48,13 @@
...
 ### JUnit tests
 
-* In Eclipse, right-click on the `test/java` folder and choose `Run as` > `JUnit Test`
+* In IntelliJ, right-click on the `test` folder and choose `Run 'All Tests'`
+```
+If `StorageFileTest` throws `NoSuchFileException`,
+1. Click the `All in test` drop down menu at the top right corner of the window
+2. Select `Edit configurations...`
+3. Under `Configuration`, change the `Working directory` to the `addressbook-level2` folder
+```
```
Comment (yamgent) [inline doc/DeveloperGuide.md]: I don't think we need the \``` here, show it as per normal (otherwise it stands out rather prominently).

**Comment by limmlingg** [comment]
> However, it is not really a "troubleshooting" because this will most likely happen to everyone.

The troubleshooting section has one where "Test fails during the very first time" which is also likely to happen to everyone.
@yamgent

**Comment by yamgent** [comment]
> The troubleshooting section has one where "Test fails during the very first time" which is also likely to happen to everyone.
> @yamgent

Ok, we will move it there then.

**Comment by CanIHasReview** [comment]
# v6

@limmlingg submitted v6 for review.

* [v6 1/1] [Convert docs from Eclipse to IntelliJ](https://github.com/se-edu/addressbook-level2/pull/201/commits/2a45b9004114fdef72b2b7c57a0eac9b8d53057e)

([:books: Archive](https://github.com/se-edu/addressbook-level2/compare/1603ad0271fc873b66ca35d4f1a922d3e199574a...2a45b9004114fdef72b2b7c57a0eac9b8d53057e)) ([:chart_with_upwards_trend: Interdiff between v5 and v6](https://github.com/se-edu/addressbook-level2/compare/9e0bb5e8e1fbf1323eb83fcfca171019e62aa8f4~1...9e0bb5e8e1fbf1323eb83fcfca171019e62aa8f4))
<details>
<summary>Checkout this PR version locally</summary>

```sh
git fetch https://github.com/se-edu/addressbook-level2.git refs/pr/201/6/head:BRANCHNAME
```
where `BRANCHNAME` is the name of the local branch you wish to fetch this PR to.

</details>

**Pull request merged**